lib/resource.js

```diff
@@ -31,11 +32,12 @@ class Resource{
 		this.name = basename(path);
 		this.consumeStats(stats);
 		
-		const repo = this.getRepository();
-		if(null !== repo){
-			this.repo = repo;
-			this.watchRepo();
-		}
+		this.getRepository().then((repo) => {
```

Comment:
~~~diff
-this.getRepository().then((repo) => {
+this.getRepository().then(repo => {
~~~